### ISSUE TITLE
.net-core WebSocket Upgrade failed with a bad request (400)

### ISSUE THREAD

**Issue by skandyrocks** [Azure-Samples/cognitive-services-speech-sdk#36]
All of the sudden I experience the following error:
`WebSocket Upgrade failed with a bad request (400). Please check the language name and deployment id, and ensure the deployment id (if used) is correctly associated with the provided subscription key.`

I provided the right subscription key and also the right location. Here's the code im using:
```
var stream = new MemoryStream(audioData);
stream.WriteWavHeader(false, 1, 16, 16000, audioData.Length / 2);

var factory = SpeechFactory.FromSubscription("key", "northeurope");
using (var recognizer = factory.CreateSpeechRecognizerWithStream(new BinaryAudioStreamReader(
    new AudioInputStreamFormat
    {
        FormatTag = 1,
        BitsPerSample = 16,
        BlockAlign = 2,
        Channels = 1,
        SamplesPerSec = 16000,
        AvgBytesPerSec = 32000
    }, stream), language))
{
    // only for single shot recognition like command or query. For long-running recognition, use
    // StartContinuousRecognitionAsync() instead.
    var recognitionResult = await recognizer.RecognizeAsync();

    ...
}
```

Am I doing something wrong? It worked two weeks ago or so
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> see above

### Any log messages given by the failure
> WebSocket Upgrade failed with a bad request (400). Please check the language name and deployment id, and ensure the deployment id (if used) is correctly associated with the provided subscription key.

### Expected/desired behavior
> Everything works fine and I get my speech recognition result

### OS and Version?
> Windows 10 (on Parallels AND also on Azure App Service)

### Versions
> SDK .net-core versions 0.5.0 (Communication Error 400) and 0.6.0 (see error message above

**Comment by wolfma61** [comment]
we found an issue between client and service that could create in rare circumstances a problem reported as an upgrade failure. this has been fixed on the service and in an upcoming sdk version (soon available)

there is the issue of concurrent connections - these are limited for the free as well as for the 'paid' keys. If you exceed your concurrent connection limit, this might currently also get reported as an upgrade issue. In the future we will have better reporting here ...

Wolfgang
